libraries/profile.rb

```diff
@@ -61,7 +62,7 @@ class ComplianceProfile < Chef::Resource # rubocop:disable Metrics/ClassLength
 
       # retrieve access token if a refresh token is set
       access_token = token
-      access_token = retrieve_access_token unless refresh_token.nil?
+      access_token = retrieve_access_token(server, refresh_token) unless refresh_token.nil?
```

Comment:
I would prefer if we would set the `node['audit']['insecure']` attribute as default to `true` and pass in the required attribute.